test/gitea_test.go

```diff
@@ -720,6 +720,79 @@ func verifyProvinance(t *testing.T, topts *tgitea.TestOpts, expectedOutput, cNam
 	assert.NilError(t, err)
 }
 
+// TestGiteaGlobalRepoConcurrencyLimit tests the concurrency_limit feature of the PipelineRun.
+// It fetches the PipelineRun definition from the default branch of the repository
+// as configured on the git platform (e.g., main).
+// In this test, the concurrency_limit is enabled using a global repository instead of a local repository.
+func TestGiteaGlobalRepoConcurrencyLimit(t *testing.T) {
+	numPipelines := 5
+	yamlFiles := map[string]string{}
+	for i := 0; i < numPipelines; i++ {
+		yamlFiles[fmt.Sprintf(".tekton/pr%d.yaml", i)] = "testdata/pipelinerun.yaml"
+	}
+	topts := &tgitea.TestOpts{
+		TargetEvent:          triggertype.PullRequest.String(),
+		YAMLFiles:            yamlFiles,
+		CheckForNumberStatus: numPipelines,
+		CheckForStatus:       "success",
+	}
+
+	verifyConcurrency(t, topts)
+}
+
+// TestGiteaGlobalAndLocalRepoConcurrencyLimit verifies the concurrency_limit feature of the PipelineRun,
+// ensuring that when concurrency_limit is defined in both global and local repository,
+// the local repository limit takes precedence. This end-to-end test confirms that behavior.
+func TestGiteaGlobalAndLocalRepoConcurrencyLimit(t *testing.T) {
+	numPipelines := 5
+	yamlFiles := map[string]string{}
+	for i := 0; i < numPipelines; i++ {
+		yamlFiles[fmt.Sprintf(".tekton/pr%d.yaml", i)] = "testdata/pipelinerun.yaml"
+	}
+	topts := &tgitea.TestOpts{
+		TargetEvent:          triggertype.PullRequest.String(),
+		YAMLFiles:            yamlFiles,
+		CheckForNumberStatus: numPipelines,
+		ConcurrencyLimit:     github.Int(3),
+		CheckForStatus:       "success",
+	}
+
+	verifyConcurrency(t, topts)
+}
+
+func verifyConcurrency(t *testing.T, topts *tgitea.TestOpts) {
+	ctx := context.Background()
```

Comment:
i think we want to move this helper method out of gitea_test.go to the pkg directory and mark it as t.Helper()